### DISCUSSION TITLE
Add photoURL to AuthUser

### DISCUSSION THREAD

**Discussion by ubresponsive** [gladly-team/next-firebase-auth#73] [Q&A] [answered]
is there a way to add PhotoURL to the AuthUser? Right now to get it I have to wait for the AuthUser.FirebaseUser to return and retrieve via AuthUser.firebaseUser.photoURL. Is there a callback to know when firebaseUser is initialised?

**Comment by kmjennison** [comment, marked as answer]
Sure, we can add a photoURL field, given it exists in both the decoded ID token and client JS:
https://firebase.google.com/docs/reference/admin/node/admin.auth.DecodedIdToken#optional-picture
https://firebase.google.com/docs/reference/js/firebase.User#photourl

Feel free to open a feature request. We could also add phone number, too.

> Is there a callback to know when firebaseUser is initialised?

There is not. I recommend using React hooks (like `useEffect`) to take some action when the Firebase JS user is ready.

**Reply by ubresponsive** [reply]
awesome, thanks for the help, I will put in a separate feature request!

**Reply by egisz** [reply]
> There is not. I recommend using React hooks (like `useEffect`) to take some action when the Firebase JS user is ready.

Hi, @kmjennison, I also have case where I would like to hook into  firebaseUser init event. I would like read user profile once logged (or read from cookie) and store it globally as UserContext.
I am currently using your suggested workaround, but I have to fetch user profile in different places, like Header, also deeper in some nested Favorite component (to check state), etc.
It would be handy to have ProfileProvider/Context and fetch once, use where ever needed.

Or, maybe there is better ways of reaching same goal?

**Reply by kmjennison** [reply]
@egisz In your case, maybe it makes sense for you to create your own context provider that extends or relies on AuthUser. Then, you can use whatever auth value you need throughout the app.
